### PR TITLE
Pytest Style: test_imports.py

### PR DESCRIPTION
**Pull request by utkbansal**
Fixes #

Changes made in this Pull Request:
 - 

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

File: testsuite/MDAnalysisTests/utils/test_imports.py [hunk 1]
```diff
@@ -22,36 +22,42 @@
...
+        for f in filter(lambda x: x.endswith('.py'), files):
+            fpath = os.path.join(dirpath, f)
+            if is_excluded(fpath):
                 continue
-            for f in filter(lambda x: x.endswith('.py'), files):
-                fpath = os.path.join(dirpath, f)
-                if self.is_excluded(fpath):
-                    continue
-                yield self._run_test_relative_import, fpath
+            # yield self._run_test_relative_import, fpath
```
Comment (orbeckst) [inline testsuite/MDAnalysisTests/utils/test_imports.py]: remove the commented line

File: testsuite/MDAnalysisTests/utils/test_imports.py [hunk 2]
```diff
@@ -22,36 +22,42 @@
...
+            paths.append(fpath)
+    return paths
+
+
+@pytest.mark.parametrize('testing_module', get_file_paths())
```
Comment (orbeckst) [inline testsuite/MDAnalysisTests/utils/test_imports.py]: looks like a sensible and pretty clean way to turn the yield-based test into a parametrized one, good!

File: testsuite/MDAnalysisTests/utils/test_imports.py [hunk 3]
```diff
@@ -22,36 +22,42 @@
...
 import os
+
 import MDAnalysisTests
+import pytest
+
+"""Relative imports are banned in unit testing modules (Issue #189), so run tests to enforce this policy."""
+
+path_to_testing_modules = MDAnalysisTests.__path__[0]
+# Exclusion path relative to MDAnalysisTests
+exclusions = ['/plugins']
```
Comment (orbeckst) [inline testsuite/MDAnalysisTests/utils/test_imports.py]: Also add `'/data'` to exclusions – will save a tiny bit in the `os.walk` and makes clear that *nothing* in the data directory should be looked at. (There are no `.py` files in `data` so nothing changes overall anyway.)

File: testsuite/MDAnalysisTests/utils/test_imports.py (hunk 2 above)
Comment (orbeckst) [inline testsuite/MDAnalysisTests/utils/test_imports.py]: (nothing to do here...)

**Comment by utkbansal** [comment]
@kain88-de @richardjgowers Review please.

File: testsuite/MDAnalysisTests/utils/test_imports.py [hunk 4]
```diff
@@ -22,36 +22,41 @@
...
 import os
+
 import MDAnalysisTests
+import pytest
+
+"""Relative imports are banned in unit testing modules (Issue #189), so run tests to enforce this policy."""
+
+path_to_testing_modules = MDAnalysisTests.__path__[0]
+# Exclusion path relative to MDAnalysisTests
+exclusions = ['/plugins', '/data']
```
Comment (jbarnoud) [inline testsuite/MDAnalysisTests/utils/test_imports.py]: Constants in the global scope should be in capital. `PATH_TO_TESTING_MODULES` and `EXCLUSIONS`. I'd like to see this changed but it is not blocking.

File: testsuite/MDAnalysisTests/utils/test_imports.py (hunk 2 above)
Comment (jbarnoud) [inline testsuite/MDAnalysisTests/utils/test_imports.py]: It is rather neat indeed!

File: testsuite/MDAnalysisTests/utils/test_imports.py (hunk 4 above)
Comment (richardjgowers) [inline testsuite/MDAnalysisTests/utils/test_imports.py]: Yeah I'd have preferred this too, it makes it obvious where things have come from

**Pull request merged**